### PR TITLE
Remove `d_prototype` & create type tests for `p_prototype`

### PR DESCRIPTION
**Pull request by SKopecz**
1. There is no need for a `d_prototype`, see the discussion in #99. In the algorithm caches we simply use `D=similar(u)` etc. A `PDSFunction` still has a field `d_protoype` which is initialized as  `d_protoype=similar(u0)`, but users cannot change this or pass `d_protoype` as a keyword to a `PDSProblem`.
2. An additional test ensures that the `P`s in the algorithm caches have the same type as a given `p_prototype` (only for in-place)

### CODE REVIEW

**Comment by SKopecz** [comment]
Does the version change to v0.2.0?

**Comment by ranocha** [comment]
> Does the version change to v0.2.0?

Yes - this is a breaking change, so we have to increment the leading non-zero digit of the version number. We should also add something like

```
## Changes when updating to v0.2 from v0.1.x

#### Added

#### Changed

#### Deprecated

#### Removed

- The optional keyword argument `d_prototype` has been removed from `PDSProblem`
```

to NEWS.md

File: test/runtests.jl [hunk 1]
```diff
@@ -1193,6 +1190,76 @@ end
             end
         end
 
+        # Here we check that the type of p_prototype actually 
+        # defines the types of the Ps inside the algorithm caches.
+        # We test sparse, tridiagonal and dense matrices as well as sparse and 
+        # dense vectors
```
Comment (ranocha) [inline test/runtests.jl:1196]:
```suggestion
        # We test sparse, tridiagonal, and dense matrices.
```

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks! Could you please increment the version number and describe the changes in NEWS.md?

**Comment by SKopecz** [comment]
This should now be ready to merge.

**Comment by ranocha** [comment]
Thanks! I would like to merge #105 first, release a new version of PositiveIntegrators.jl, and merge this PR afterwards

**Review by ranocha — APPROVED** [review]

**Pull request merged**